src/main/java/io/penguinstats/model/Stage.java

```diff
@@ -22,20 +24,48 @@
 @JsonInclude(JsonInclude.Include.NON_NULL)
 public class Stage implements Serializable {
 
+	public interface StageBaseView {};
+
+	public interface StageNewView extends StageBaseView {};
+
+	public interface StageLegacyView extends StageBaseView {};
```

Comment:
The view annotations are messy...
How about we define some interfaces, like NewView<T>, with public function `T toNewView()`
Then in model classes, implements NewView and override the function like this:
```
override function Stage toNewView() {
    this.stageType = null;
    this.stageId = null;
   ...
   return this;
}
```
And the null value will not appear in the json.

Reply:
fixed